torchaudio/csrc/rnnt/cpu/compute.cpp

```diff
@@ -87,10 +87,7 @@ std::tuple<torch::Tensor, c10::optional<torch::Tensor>> compute(
   torch::Tensor costs = torch::empty(
       options.batchSize_ * options.nHypos_,
       torch::TensorOptions().device(logits.device()).dtype(logits.dtype()));
-  c10::optional<torch::Tensor> gradients = c10::nullopt;
-  if (logits.requires_grad()) {
-    gradients = torch::zeros_like(logits);
-  }
+  c10::optional<torch::Tensor> gradients = torch::zeros_like(logits);
```

Comment:
Here, Tensor with the same number of elements as `logits` is initialized, even when `requires_grad=False`.

Looking at all the cases, the storage of the tensor is passed to `Compute` function.
Can we handle  the case `c10::optional<torch::Tensor> gradients == nullopt` in `Compute` function, ideally, without allocating `gradient` tensor?

Reply:
It is eventually passed to https://github.com/pytorch/audio/blob/8a347b62cf5c907d2676bdc983354834e500a282/torchaudio/csrc/rnnt/cpu/cpu_kernels.h#L373

and it seems to me not easy to fix this function. So I think the current approach of quick fix is good enough. Maybe we can file the issue for the further improvement (handle `gradient==nullptr` in `ComputeGradient`) in the future.